### ISSUE TITLE
Describe accounting terms and systems

### ISSUE THREAD

**Issue by Adzox** [frivara/accounting#4]
Describe how the accounting system works, what it consists of and how to work it.

Also make a small list of terms and words and what they mean.

**Comment by Adzox** [comment]
Terms and words:

* Double-entry bookkeeping: every transaction is recorded in at least two accounts
* Transaction (Transaktion): Money you've gotten or lost. Must  registered either as debit or credit. Debit and credit must total 0 for a transaction to be valid.
    * Example: We got money from a new member, which paid for both training and membership in one payment. The transaction could look like the following:
    1910 Kassa: 1050 SEK Debit
    3990 Träningsavgift: 1000 SEK Credit
    3991 Medlemsskapsavgift: 50 SEK Credit 
* Debit: Money gained in one account - think plus
* Credit: Money lost in one account - think minus
* Verification (Verifikation): Document proving a transaction actually occurred - think receipts and invoices
* Account: A four-digit number (and name) identifying one group or kind of expense or income. Example is 1920, which
* Chart of Accounts (Kontoplan): All accounts used in the bookkeeping, with four-digit code and name.
* BAS: An organization and their template Chart of Accounts
* Idrottens BAS-Kontoplan: A template Chart of Accounts set up by Riksidrottsförbundet

* Fiscal year: The time span used when bookkeeping - does not have to be YYYY-01-01 - YYYY-12-31, but often is
* Financial Statements/Report (Bokslut): Reports summarizing the bookkeepings for a fiscal year, consisting of a Balance Report and a Result Report
* Balance Report: Report detailing the financial position at a point in time of the organization
* Result Report: Report detailing the results of a year, whether that be gains or losses
* Kontometoden/Fakturametoden: Way of bookkeeping, see https://www.bokio.se/hjalp/komma-igang/bokforing/vilken-bokforingsmetod-ska-jag-anvanda/?psafe_param=1&gad=1&gclid=CjwKCAjwpJWoBhA8EiwAHZFzfjQJNXyCDFiryizE-jkb9j7c2Z4o2nXNBuR29RK2a2vHf-TnmgBWPBoC3D8QAvD_BwE

Soon, I will detail how I've worked with accounting and bookkeeping

**Comment by Adzox** [comment]
When starting accounting, some base information needs to be provided if the organization has already done bookkeeping. Usually you would import your bookkeeping information from your previous software, saved in a SIE format, but some software also have a manual way of getting started. Either way, what is inputted is the starting balance of each account, so how much is in them from start, the duration that the bookkeeping should cover (fiscal year) and the Chart of Accounts to use for the bookkeeping.

When bookkeeping, you will create new transactions into a fiscal year that is open (i.e. it hasn't been finalized/isn't completed), uploading verifications to those transactions and selecting accounts from the Chart of Accounts to balance it with.

Very important point is that a transaction, once saved and added to the fiscal year, __is not editable or deletable__ - rather, if a verification needs to be revoked or edited, a new transaction is made specifically to revert and correct a previous one. These will officially be counted as two different transactions, but most software shows their connection in some manner. This is a legal requirement for electronical bookkeeping systems in Sweden to be counted as a valid program to use for bookkeeping.

Once all transactions has been made for the fiscal year, and the accountant has checked that everything is correct, it is time to close the bookkeeping. This process entails creating some final transactions and locking the year. This is a step which differs depending on what kind of organization it is, so to keep it simple for now, this step will be kept manual. All that remains is therefore to lock the  year, making it non-editable and finalized. Some software also allow you to unlock a year and make edits, but require that the previous bookkeeping has been closed and locked before starting a near year.

When creating a new year, the input settings needed are fetched from the previous fiscal year's bookkeeping, making the process smoother.

When closing the books and when the books has been closed, that is when Balance Reports and Result Reports are generated - before as a preview to verify the accounting looks OK, and after to create the "official" reports.

**Comment by Adzox** [comment]
@AlexanderTarler please have a look at this and see if it makes sense!

**Comment by Adzox** [comment]
Note: our system will probably only support Kontantmetoden at first, if the Fakturametoden is too different or requires additional features.

**Comment by AlexanderTarler** [comment]
I'm late to the party, but this looks great! Simple yet efficient. Well done!
